tests/neynarAPI.ts

```diff
@@ -0,0 +1,609 @@
+import chai, { expect } from "chai";
+import chaiAsPromised from "chai-as-promised";
+import { expectDefined } from "./utils";
+import {
+  NeynarAPIClient,
+  Logger,
+  silentLogger,
+  generateSignature,
+} from "../src/";
+import {
+  Cast as v2Cast,
+  PostCastResponseCast,
+  EmbedUrl,
+  ReactionType,
+} from "../src/neynarAPI/neynarV2API";
+import { ReactionTypeEnum } from "../src/neynarAPI/neynarV1API";
+
+chai.use(chaiAsPromised);
+
+const privateKey = process.env.INTEGRATION_TEST_USER_MNEMONIC;
+const apiKey = process.env.INTEGRATION_TEST_NEYNAR_API_KEY;
+const signerUuid = process.env.INTEGRATION_TEST_NEYNAR_SIGNER_UUID;
+const signerPublicKey = process.env.INTEGRATION_TEST_NEYNAR_SIGNER_PUBLIC_KEY;
+
+const testLogger: Logger = {
+  info: silentLogger.info,
+  debug: silentLogger.debug,
+  trace: silentLogger.trace,
+  warn: silentLogger.warn,
+
+  /* eslint-disable no-console */
+  error: console.error,
+  /* eslint-enable no-console */
+};
+
+const userDwrFid = 3;
+const userGaviFid = 69; // @gavi
+const userBotFid = 6365; // @gavi-bot
+
+async function sleep(ms: number): Promise<void> {
+  return await new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+if (apiKey !== undefined && apiKey !== "") {
+  describe("NeynarAPI", function () {
+    this.timeout("30000");
+
+    let client: NeynarAPIClient;
+
+    before("create client", function () {
+      client = new NeynarAPIClient(apiKey, { logger: testLogger });
+    });
+
+    describe("#signer", function () {
+      /**
+       * Note: While testing please reuse the signer, it costs money to approve a signer.
+       * Steps to create & register a signer:
+       * 1. Remove the skip on "can create a signer" and capture the Signer UUID & public key from the logs
+       * 2. Set the values for your Signer UUID & public key in your .env file
+       * 3. Replace the skip to "can create a signer" and remove the skip on "can register a signer"
+       *   - Update the deadline if you want to use the signer for a longer period
+       * 4. Open the signer_approval_url from the logged registeredSigner on a mobile device
+       *    that is logged into the same account
+       *    - if android -> update the deeplink to follow this format
+       *     https://client.warpcast.com/deeplinks/signed-key-request?token=0x1234
+       * 5. Approve the signer (costs $0.99)
+       * 6. Replace the skip on "can register a signer" and run the tests
+       */
```

Comment:
This looks like it might be better suited in the neynar-specific README's "contributing" section

Reply:
Want to just cut-paste this over to CONTRIBUTING.md now that we have it?